src/VTUFiles.f90

```diff
@@ -1,643 +1,497 @@
+        IF(PRESENT(recl)) CALL fileobj%e%raiseDebug(modName//'::'//myName// &
+          ' - Optional input "RECL" is being ignored. File is "SEQUENTIAL".')
+        !
+        !Initialize the file, only support ascii for now
+        CALL init_fortran_file(fileobj,unit,file,'REPLACE','SEQUENTIAL', &
+          'FORMATTED','APPEND','WRITE')
+        !
+        !Open the file
+        CALL fileobj%fopen()
+        !
+        !Handle status argument, which is unused
+        IF(PRESENT(status)) THEN
+          CALL fileobj%e%raiseDebug(modName//'::'//myName// &
+            ' - File header not used for VTU files!')
+        ENDIF
+        !
+        !Write header to file, title and format information to file
+        IF(fileobj%isOpen()) THEN
+          WRITE(fileobj%getUnitNo(),'(a)') '<?xml version="1.0"?>'
+          WRITE(fileobj%getUnitNo(),'(a)') '<VTKFile type='// &
+          '"UnstructuredGrid" version="0.1" byte_order="LittleEndian">'
+          !Change "new" status so that if file is closed and re-opened
+          !during execution it will be appended file instead of replacing it.
+          fileobj%newstat=.FALSE.
+        ENDIF
+      ELSE
+        CALL fileobj%e%raiseError(modName//'::'//myName// &
+          ' - VTU File is already initialized!')
+      ENDIF
+    ENDSUBROUTINE init_VTUXMLFileType
+!
+!-------------------------------------------------------------------------------
+!> @brief Writes a VTU mesh to VTU file
+!> @param myVTKFile the VTK file to write the data to
+!> @param vtkMesh the VTK mesh to write in the file
+!> 
+!> This routine supports writing of UNSTRUCTURED_GRIDS. The other types of 
+!> geometry/topology are not yet implemented. The display of this 
+!> data was verified by loading and displaying it with VisIt.
+!>
+    SUBROUTINE writeMesh_VTUXMLFileType(myVTKFile,vtkMesh)
+      CHARACTER(LEN=*),PARAMETER :: myName='writeMesh_VTUXMLFileType'
+      CLASS(VTUXMLFileType),INTENT(INOUT) :: myVTKFile
+      TYPE(VTKMeshType),INTENT(IN) :: vtkMesh
+      CHARACTER(LEN=256) :: aline,sint
+      INTEGER(SIK) :: funit,i,j,k,n
+      INTEGER(SIK),DIMENSION(:),ALLOCATABLE :: offsets
+      !
+      IF(.NOT.myVTKFile%hasMesh) THEN
+        IF(myVTKFile%isOpen()) THEN
+          IF(vtkMesh%isInit) THEN
+            funit=myVTKFile%getUnitNo()
+            SELECTCASE(vtkMesh%meshType)
+              CASE(VTK_UNSTRUCTURED_GRID)
+                myVTKFile%mesh=vtkMesh
+                myVTKFile%hasMesh=.TRUE.
+                !Write a mesh that is an unstructured grid
+                !Clean-up redundant points
+                CALL myVTKFile%mesh%cleanupPoints()
+                WRITE(sint,'(i64)') myVTKFile%mesh%numPoints; sint=ADJUSTL(sint)
+                WRITE(aline,'(i64)') myVTKFile%mesh%numCells; aline=ADJUSTL(aline)
+                WRITE(funit,'(a)') '  <UnstructuredGrid>'
+                WRITE(funit,'(a)') '    <Piece NumberOfPoints="'//TRIM(sint)//'" NumberOfCells="'//TRIM(aline)//'">'
```

Comment:
Another formatting guideline is to try to keep lines from extending beyond 80-columns. The purpose of this guideline is to help with readability.

Consider continuing some source lines across multiple lines. e.g.
```
WRITE(*,*) "STuff", &
  "indent 2-spaces continuing to subsequent line"
```